### PR TITLE
fix(events): fix bugs in the event stream and improve tests

### PR DESCRIPTION
**Pull request by glevco**
### Acceptance Criteria

- Add missing docstrings and typings
- Fix parsing bug in `EventData` and add tests for it
- Fix edge cases bugs in `EventWebsocketProtocol` and add tests for them
- Implement `BaseEventSimulationTester` to help writing event queue feature tests, and new tests using it

### CODE REVIEW

File: hathor/simulator/simulator.py [hunk 1]
```diff
@@ -151,7 +152,8 @@ def create_peer(
         soft_voided_tx_ids: Optional[Set[bytes]] = None,
         full_verification: bool = True,
         event_ws_factory: Optional[EventWebsocketFactory] = None,
-        tx_storage: Optional[TransactionStorage] = None
+        tx_storage: Optional[TransactionStorage] = None,
+        event_storage: Optional[EventStorage] = None,
```
Comment (msbrogli) [inline hathor/simulator/simulator.py:156]: Why `event_storage` instead of `EventManager`?

File: tests/event/event_simulation_tester.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+from hathor.storage import RocksDBStorage
+from hathor.transaction.storage import TransactionStorage
+from hathor.transaction.util import unpack, unpack_len
+from hathor.util import json_loadb
+from tests.simulation.base import SimulatorTestCase
+from tests.utils import HAS_ROCKSDB
+
+
+class BaseEventSimulationTester(SimulatorTestCase):
+    seed_config = 6946502462188444706
```
Comment (msbrogli) [inline tests/event/event_simulation_tester.py:35]: Why are we fixing a seed? Shouldn't we let random tests run?

File: hathor/event/event_manager.py [hunk 3]
```diff
@@ -209,20 +221,24 @@ def _create_event(
...
+        """Event specific handler for EventType.LOAD_FINISHED."""
         self._event_storage.save_node_state(NodeState.SYNC)
 
     def _should_reload_events(self) -> bool:
+        """Returns whether events should be reloaded or not."""
         return self._previous_node_state in [None, NodeState.LOAD]
 
     def get_event_queue_state(self) -> bool:
-        """Get whether the event queue feature is enabled from the storage"""
+        """Gets whether the event queue feature is enabled from the storage."""
```
Comment (msbrogli) [inline hathor/event/event_manager.py:237]: Convention is to use the imperative on docstring. So, `Get whether ... `.

**Review by msbrogli — DISMISSED** [review]

File: hathor/simulator/simulator.py (hunk 1 above)
Comment (glevco) [reply hathor/simulator/simulator.py]: This has been removed after rebasing

File: tests/event/event_simulation_tester.py (hunk 2 above)
Comment (glevco) [reply tests/event/event_simulation_tester.py]: Removed in https://github.com/HathorNetwork/hathor-core/pull/647/commits/99cf418bb73591c67e42fd0520e7cd2ac94288e8

File: hathor/event/event_manager.py (hunk 3 above)
Comment (glevco) [reply hathor/event/event_manager.py]: Really? I always thought it was the opposite, there's a lot of code using the descriptive version, just search for "returns"... But I changed this in https://github.com/HathorNetwork/hathor-core/pull/647/commits/99cf418bb73591c67e42fd0520e7cd2ac94288e8

**Review by jansegre — APPROVED** [review]
Just a small note, we normally use `fix` instead of `bugfix` for both branch prefix and the commit message.

File: hathor/simulator/simulator.py (hunk 1 above)
Comment (msbrogli) [reply hathor/simulator/simulator.py]: Ok. Thanks!

File: tests/event/event_simulation_tester.py (hunk 2 above)
Comment (msbrogli) [reply tests/event/event_simulation_tester.py]: Ok. Thanks!

File: hathor/event/event_manager.py (hunk 3 above)
Comment (msbrogli) [reply hathor/event/event_manager.py]: Our docstrings are far from the convention. Anyway, thanks for changing.

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]
